golang/go-cloud-run-hello-world/main.go

```diff
@@ -0,0 +1,101 @@
+// Copyright 2019 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package main
+
+import (
+	"html/template"
+	"log"
+	"net/http"
+	"os"
+
+	"cloud.google.com/go/compute/metadata"
+)
+
+// templateData provides template parameters.
+type templateData struct {
+	Service      string
+	Revision     string
+	Project      string
+	ProjectFound bool
+}
+
+// Variables used to generate the HTML page.
+var (
+	data templateData
+	tmpl *template.Template
+)
+
+func main() {
+	// Comment out in local development for timestamp prefixes.
+	log.SetFlags(0)
+
+	// Initialize template parameters.
+	service := os.Getenv("K_SERVICE")
+	if service == "" {
+		service = "???"
+	}
+
+	revision := os.Getenv("K_REVISION")
+	if revision == "" {
+		revision = "???"
+	}
+
+	project := "???"
+	projectFound := false
+	// Only attempt to check the Cloud Run metadata server if it looks like
+	// the service is deployed to Cloud Run.
+	if service != "???" {
+		var err error
+		if project, err = metadata.ProjectID(); err != nil {
+			log.Printf("metadata.ProjectID: Cloud Run metadata server: %v", err)
+		}
+	}
+
+	// Prepare template for execution.
+	tmpl = template.Must(template.ParseFiles("index.html"))
+	data = templateData{
+		Service:      service,
+		Revision:     revision,
+		Project:      project,
+		ProjectFound: projectFound,
+	}
+
+	// Define HTTP server.
+	http.HandleFunc("/", helloRunHandler)
+
+	fs := http.FileServer(http.Dir("./assets"))
+	http.Handle("/assets/", http.StripPrefix("/assets/", fs))
+
+	// PORT environment variable is provided by Cloud Run.
+	port := os.Getenv("PORT")
+	if port == "" {
+		port = "8080"
```

Comment:
Do we need to mention in the readme that it runs on 8080 when deployed locally?

Reply:
8080 is the default and the port is output as part of server start. That seems sufficient to me unless the goal is to document the availability of the PORT var to avoid port conflicts? If so, how does that relate to only using Cloud Code command palette to run locally?

Reply:
In case developers run multiple workloads in containers locally, we may want to let them know that this sample is using $PORT to expose the service locally.

Reply:
Added README section for `PORT` and `GOOGLE_CLOUD_PROJECT`